### PR TITLE
IDVA5-499-500-correspondence-address-manual-and-confirm

### PR DESCRIPTION
**Pull request by abhiahirv1**
Correspondence Address Manual and Confirm Screen

### CODE REVIEW

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: src/test/src/controllers/unicorporatedCorrespondenceAddressConfirmController.test.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+jest.mock("@companieshouse/api-sdk-node");
+const router = supertest(app);
+
+describe("GET" + UNINCORPORATED_CORRESPONDENCE_ADDRESS_CONFIRM, () => {
+    it("should render the confirmation page with status 200", async () => {
+        const res = await router.get(BASE_URL + UNINCORPORATED_CORRESPONDENCE_ADDRESS_CONFIRM);
+        expect(res.status).toBe(200);
+        expect(mocks.mockSessionMiddleware).toHaveBeenCalled();
+        expect(mocks.mockAuthenticationMiddleware).toHaveBeenCalled();
+    });
```
Comment (ttingle-ch) [inline src/test/src/controllers/unicorporatedCorrespondenceAddressConfirmController.test.ts:16]: In the test you should check the loaded page is correct using expect(res.text).toContain(Page Heading)

File: src/test/src/controllers/unicorporatedCorrespondenceAddressConfirmController.test.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+    it("should render the confirmation page with user data", async () => {
+        const userSession = { firstName: "John", lastName: "Doe", correspondenceAddress: "123 Main St" };
```
Comment (ttingle-ch) [inline src/test/src/controllers/unicorporatedCorrespondenceAddressConfirmController.test.ts:19]: First and last name will not show on unincorporated view

File: src/test/src/controllers/unincorporatedCorrespondenceAddressManualController.test.ts [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+    xit("should return status 302 after redirect", async () => {
+        await router.post(BASE_URL + UNINCORPORATED_CORRESPONDENCE_ADDRESS_MANUAL)
+            .send({ addressPropertyDetails: "abc", addressLine1: "pqr", addressLine2: "pqr", addressTown: "lmn", addressCounty: "lmnop", addressCountry: "lmnop", addressPostcode: "MK9 3GB" }).expect(302);
+    });
+});
+// Test for no addressPropertyDetails, will return 400.
+describe("POST" + UNINCORPORATED_CORRESPONDENCE_ADDRESS_MANUAL, () => {
+    it("should return status 400", async () => {
+        await router.post(BASE_URL + UNINCORPORATED_CORRESPONDENCE_ADDRESS_MANUAL)
+            .send({ addressPropertyDetails: "", addressLine1: "pqr", addressLine2: "pqr", addressTown: "lmn", addressCounty: "lmnop", addressCountry: "lmnop", addressPostcode: "MK9 3GB" }).expect(400);
```
Comment (ttingle-ch) [inline src/test/src/controllers/unincorporatedCorrespondenceAddressManualController.test.ts:28]: add an expect to test the error message shows what is expected for each of the error scenarios

**Review by ttingle-ch — CHANGES_REQUESTED** [review]

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: src/main/controllers/features/unincorporated/unicorporatedCorrespondenceAddressConfirmController.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import { NextFunction, Request, Response } from "express";
```
Comment (swalke-ch) [inline src/main/controllers/features/unincorporated/unicorporatedCorrespondenceAddressConfirmController.ts:1]: Can you remove  unicorporated from unicorporatedCorrespondenceAddressConfirmController.ts name please? Its already part of  unicorporated feature so no need to add it in name again

File: src/main/controllers/features/unincorporated/unincorporatedCorrespondenceAddressManualController.ts [hunk 5]
```diff
@@ -0,0 +1,58 @@
+import { NextFunction, Request, Response } from "express";
```
Comment (swalke-ch) [inline src/main/controllers/features/unincorporated/unincorporatedCorrespondenceAddressManualController.ts:1]: same as above

File: src/test/src/controllers/unincorporatedCorrespondenceAddressManualController.test.ts [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+// Test for incorrect addressCounty Length entered, will return 400.
+describe("POST" + UNINCORPORATED_CORRESPONDENCE_ADDRESS_MANUAL, () => {
+    it("should return status 400", async () => {
+        const res = await router.post(BASE_URL + UNINCORPORATED_CORRESPONDENCE_ADDRESS_MANUAL)
+            .send({ addressPropertyDetails: "abc", addressLine1: "pqr", addressLine2: "abc", addressTown: "abc", addressCounty: "Abcdefghijklmnopqrstuvwx1Abcdefghijklmnopqrstuvwx2A", addressCountry: "lmnop", addressPostcode: "MK9 3GB" }).expect(400);
+        expect(res.text).toContain("County or region must be 50 characters or less");
+    });
+});
+
+// Test for no addressCountry, will return 302.
```
Comment (ttingle-ch) [inline src/test/src/controllers/unincorporatedCorrespondenceAddressManualController.test.ts:149]: Country is a mandatory field so should return 400 if no country is provided

File: src/test/src/controllers/unincorporatedCorrespondenceAddressManualController.test.ts [hunk 7]
```diff
@@ -0,0 +1,194 @@
...
+            .send({ addressPropertyDetails: "abc", addressLine1: "abc", addressLine2: "abc", addressTown: "abc", addressCounty: "abcop", addressCountry: "abcop", addressPostcode: "" }).expect(400);
+        expect(res.text).toContain("Enter a postcode");
+    });
+});
+// Test for incorrect addressPostcode Format entered, will return 400.
+describe("POST" + UNINCORPORATED_CORRESPONDENCE_ADDRESS_MANUAL, () => {
+    it("should return status 400", async () => {
+        await router.post(BASE_URL + UNINCORPORATED_CORRESPONDENCE_ADDRESS_MANUAL)
+            .send({ addressPropertyDetails: "abc", addressLine1: "pqr", addressLine2: "abc", addressTown: "lmn", addressCounty: "lmnop", addressCountry: "lmnop", addressPostcode: "MK9 3GB@" }).expect(400);
+    });
```
Comment (ttingle-ch) [inline src/test/src/controllers/unincorporatedCorrespondenceAddressManualController.test.ts:186]: Missing expected text

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Review by ttingle-ch — APPROVED** [review]

**Review by swalke-ch — APPROVED** [review]

**Pull request merged**